codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -479,4 +274,158 @@ public HttpHeaders clear() {
         super.clear();
         return this;
     }
+
+    @Override
+    public int hashCode() {
+        return size();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof DefaultHttpHeaders)) {
```

Comment:
I wonder if we should check against HttpHeaders and not against the actual implementation.